### PR TITLE
add sttp-oauth2-cache-cats module with CE3 support

### PR DESCRIPTION
**Pull request by bwiercinski**
aims https://github.com/ocadotechnology/sttp-oauth2/issues/172

### CODE REVIEW

File: oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      for {
+        cache <- CatsRefExpiringCache[IO, String, Int]
+        now   <- IO.realTimeInstant
+        _     <- cache.put(someKey, someValue, now.plusSeconds(60))
+        _     <- IO.sleep(60.seconds - 1.nano)
+        value <- cache.get(someKey)
+      } yield value
+    }.shouldBe(Succeeded(Some(Some(someValue))))
+
+    "not return value if expired" in unsafeRun {
```
Comment (matwojcik) [inline oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala:52]: There is removed test `remove value on expired get` from CE2 tests, why?

**Review by matwojcik — APPROVED** [review]

File: oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala (hunk 1 above)
Comment (bwiercinski) [reply oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala:52]: oh right, i wanted to mention it. it is impossible to make this test using `Ticked`. i'm getting assertion errors when i try to sleep with negative duration

also using TestControl, it seems not working. you can play with my scratch by yourself

```scala
"remove value on expired get" in TestControl
  .execute(for {
    cache  <- CatsRefExpiringCache[IO, String, Int]
    now    <- IO.realTimeInstant
    _      <- cache.put(someKey, someValue, now.plusSeconds(60))
    _      <- /* Bookmark 1 */ IO.sleep(60.seconds)
    value1 <- cache.get(someKey) // this call should remove expired value from cache
    _      <- /* Bookmark 2 */ IO.sleep(1.seconds)
    value2 <- cache.get(someKey)
  } yield (value1, value2))
  .flatMap { control =>
    for {
      _      <- control.tick // go to: Bookmark 1
      a      <- control.nextInterval
      _      <- control.advanceAndTick(a)  // go to: Bookmark 2
      b      <- control.nextInterval
      _      <- control.advanceAndTick(-10 * b)
      result <- control.results
    } yield result shouldBe Some(Succeeded((None, None)))
  }
  .unsafeRunSync()(cats.effect.unsafe.implicits.global)
```

Comment (matwojcik) [reply oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala:52]: So maybe create ref inside test, use it to create `CatsRefExpiringCache` instance, and then after expired get peek into the ref to see if the entry is removed?

Comment (bwiercinski) [reply oauth2-cache-cats/src/test/scala/com/ocadotechnology/sttp/oauth2/cache/cats/CatsRefExpiringCacheSpec.scala:53]: https://github.com/ocadotechnology/sttp-oauth2/pull/173/commits/d869ca83939951fe1b0f0f61d5250d3da4b85488

**Review by majk-p — APPROVED** [review]

**Pull request merged**